packages/validator-query/lib/AmpValidatorQuery.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class AmpValidatorQuery {
+  constructor(rules) {
+    this.initRules_(rules);
+  }
+
+  getTagsForFormat(format, transformed) {
+    format = format.toLowerCase();
+    return this.tags
+        .filter(
+            (tag) =>
+              tag.htmlFormat.includes(format.toUpperCase()) &&
+          this.checkEntityFormat_(tag, format) &&
+          this.checkEntityTransformed_(tag, transformed)
+        )
+        .map((tag) => {
+          tag = Object.assign({}, tag);
+          tag.attrs = tag.attrs.filter(
+              (attr) =>
+                this.checkEntityFormat_(attr, format) &&
+            this.checkEntityTransformed_(attr, transformed)
+          );
+          return tag;
+        });
+  }
+
+  getExtensionsForFormat(format) {
+    format = format.toUpperCase();
+    return this.extensions
+        .filter((extension) => extension.htmlFormat.includes(format))
+        .reduce((result, extension) => {
+          result[extension.name] = Object.assign({}, extension);
+          delete result[extension.name].name;
+          return result;
+        }, {});
+  }
+
+  checkEntityTransformed_(entity, transformed) {
+    if (transformed) {
+      return this.checkEntityFormat_(entity, 'transformed');
+    }
+    if (entity.enabledBy && entity.enabledBy.includes('transformed')) {
```

Comment:
Suggestion: extract this part into a helper method:

```
if (entity.enabledBy && entity.enabledBy.includes('transformed')) {
       return false;
     }
     if (entity.disabledBy && !entity.disabledBy.includes('transformed')) {
       return false;
     }
```

Reply:
Looking back, I wonder if I can just reuse `checkEntityFormat_` and call it as `!this.checkEntityFormat_(entity, 'transformed')`...

Will need to give it some thought.